sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlInferSchema.scala

```diff
@@ -130,10 +147,28 @@ class XmlInferSchema(options: XmlOptions, caseSensitive: Boolean)
       parser.close()
```

Comment:
didn't take a super close look but should we maybe close the `parser` with `finally`?